app/controllers/publishers/vacancies/base_controller.rb

```diff
@@ -22,7 +22,10 @@ def vacancies
 
   def vacancy
     # Scope to internal vacancies to disallow editing of external ones
-    @vacancy ||= vacancies.internal.find(params[:job_id].presence || params[:id])
+
+    # As the vacancy is not associated with an organisation upon creation, calling the vacancies method will return an empty array as an organisation is not associated
+    # with it. To fix this, before the vacancy's status is set (and therefore before an organisation is associated), we find the job from the vacancies where status is nil.
```

Comment:
I'm not sure I understand why we need `where(status: nil)` 

Would `Vacancy.find(params[:job_id].presence || params[:id]))` not work?
So:

```
@vacancy ||= (vacancies.internal.find_by(id: params[:job_id].presence || params[:id]) || Vacancy.find(params[:job_id].presence || params[:id]))
```